File: Lab5/C++/Lab5.2/main2.cpp
#include "main2.h"

MainClass::Square::Square(double Width)
{
  this->Width = Width;
}

double MainClass::Square::Area()
{
  return Width * Width;
}

double MainClass::Square::Perimeter()
{
  return 4 * Width;
}

MainClass::Circle::Circle(double Radius)
{
  this->Radius = Radius;
}

double MainClass::Circle::Area()
{
  return M_PI * Radius * Radius;
}

double MainClass::Circle::Perimeter()
{
  return 2 * M_PI * Radius;
}

void MainClass::Main(std::vector<std::wstring> &args)
{
  std::wcout << L"hello world" << std::endl;

  Shape *shape;

  shape = new Square(5);
  std::wcout << L"square .Area(): " << shape->Area() << std::endl;
  std::wcout << L"square .Perimeter(): " << shape->Perimeter() << std::endl;

  shape = new Circle(5);
  std::wcout << L"circle .Area(): " << shape->Area() << std::endl;
  std::wcout << L"circle .Perimeter(): " << shape->Perimeter() << std::endl;

	delete shape;
}

File: Lab5/C#/main2.cs
using System;

class MainClass {

  abstract class Shape {
    public abstract double Area();

    public abstract double Perimeter();
  }

  class Square : Shape {
    private double Width;

    public Square(double Width) {
      this.Width = Width;
    }

    public override double Area() {
      return Width * Width;
    }

    public override double Perimeter() {
      return 4 * Width;
    }
  }

  class Circle : Shape {
    private double Radius;

    public Circle(double Radius) {
      this.Radius = Radius;
    }

    public override double Area() {
      return Math.PI * Radius * Radius;
    }

    public override double Perimeter() {
      return 2 * Math.PI * Radius;
    }
  }

  public static void Main(string[] args) {
    Console.WriteLine("hello world");

    Shape shape;

    shape = new Square(5);
    Console.WriteLine("square .Area(): " + shape.Area());
    Console.WriteLine("square .Perimeter(): " + shape.Perimeter());

    shape = new Circle(5);
    Console.WriteLine("circle .Area(): " + shape.Area());
    Console.WriteLine("circle .Perimeter(): " + shape.Perimeter());
  }
}
